src/Draco.LanguageServer/Capabilities/GotoDefinition.cs

```diff
@@ -30,14 +30,17 @@ public Task<IList<Location>> GotoDefinitionAsync(DefinitionParams param, Cancell
             .Select(semanticModel.GetReferencedSymbol)
             .LastOrDefault(symbol => symbol is not null);
 
-        if (referencedSymbol is not null && referencedSymbol.Definition is not null)
-        {
-            var location = Translator.ToLsp(referencedSymbol.Definition);
-            return Task.FromResult<IList<Location>>([location ?? default!]);
-        }
-        else
+        if (referencedSymbol is null) return Task.FromResult<IList<Location>>([]);
+
+        if (referencedSymbol.IsGenericInstance)
         {
-            return Task.FromResult<IList<Location>>(Array.Empty<Location>());
+            // Unwrap to resolve to the generic definition
+            referencedSymbol = referencedSymbol.GenericDefinition;
         }
+
+        if (referencedSymbol.Definition is null) return Task.FromResult<IList<Location>>([]);
+
+        var location = Translator.ToLsp(referencedSymbol.Definition);
+        return Task.FromResult<IList<Location>>([location ?? default!]);
```

Comment:
I didn't saw this before, but avoid returning a Task, prefer making method async and await.